src/diff/oasDiff.ts

```diff
@@ -14,71 +189,327 @@ import { execSync } from "child_process";
+ * @returns The stdout output from oasdiff
+ */
+async function executeOasDiff(
+  baseSpec: string,
+  newSpec: string,
+  jsonMode: string,
+  directoryMode = ""
+): Promise<string> {
+  return new Promise((resolve, reject) => {
+    const flags = [jsonMode, directoryMode]
+      .filter((flag) => flag.trim() !== "")
+      .join(" ");
+    const flagsString = flags ? ` ${flags}` : "";
+    // intentionally not leaving space for flagsString
+    exec(
+      `oasdiff changelog${flagsString} "${baseSpec}" "${newSpec}"`,
+      (error, stdout) => {
+        if (error) {
+          reject(error);
+        } else {
+          resolve(stdout);
+        }
+      }
+    );
+  });
+}
+
+/**
+ * Handle directory mode comparison logic
+ *
+ * @param baseApi - The base API directory
+ * @param newApi - The new API directory
+ * @param jsonMode - JSON formatting flag
+ * @param flags - CLI flags
+ * @returns Object containing results, hasChanges flag, and hasErrors flag
+ */
+async function handleDirectoryMode(
+  baseApi: string,
+  newApi: string,
+  jsonMode: string,
+  flags: OasDiffFlags
+): Promise<OasDiffHandlerResult> {
+  const allResults: OasDiffResult[] = [];
+  let hasChanges = false;
+  let hasErrors = false;
+
+  // Find all exchange.json files and their parent directories
+  const baseExchangeDirs = await findExchangeDirectories(baseApi);
+  const newExchangeDirs = await findExchangeDirectories(newApi);
+
+  const allDirNames = new Set([
+    ...baseExchangeDirs.map((dir) => dir.name),
+    ...newExchangeDirs.map((dir) => dir.name),
+  ]);
+
+  for (const dirName of allDirNames) {
+    const baseDir = baseExchangeDirs.find((dir) => dir.name === dirName);
+    const newDir = newExchangeDirs.find((dir) => dir.name === dirName);
+
+    // Check if directory was deleted
+    if (baseDir && !newDir) {
+      console.log(`${dirName} API is deleted`);
+      if (flags.format === "json") {
+        allResults.push({
+          directory: dirName,
+          status: "deleted",
+          message: `${dirName} API is deleted`,
+        });
+      } else {
+        allResults.push(`======${dirName} API is deleted======`);
+      }
+      hasChanges = true;
+      continue;
+    }
+
+    // Check if directory was added
+    if (!baseDir && newDir) {
+      console.log(`${dirName} API is added`);
+      if (flags.format === "json") {
+        allResults.push({
+          directory: dirName,
+          status: "added",
+          message: `${dirName} API is added`,
+        });
+      } else {
+        allResults.push(`======${dirName} API is added======`);
+      }
+      hasChanges = true;
+      continue;
+    }
+
+    // Both directories exist, compare individual yml files in each directory
+    if (baseDir && newDir) {
+      console.log("===================================");
+      console.log(`Processing directory pair: ${dirName}`);
+
+      try {
+        const baseYamlFiles = await findYamlFiles(baseDir.path);
```

Comment:
Multiple yaml files? Is this because we now have internal and public yaml files?